components/search/README.md

```diff
@@ -0,0 +1,68 @@
+
+It is currently under construction and not yet used.
+
+## Architecture
+
+This module currently has one function, `filter_engine_configuration`, which receives configurations from remote settings and details of the user's environment. It returns a list of search engines to be presented to the user.
+
+This component follows the architecture of the other [Application Services Rust components](https://mozilla.github.io/application-services/book/index.html): a cross-platform Rust core, and platform-specific bindings for Firefox Desktop, Android, and iOS. These bindings are generated automatically using the [UniFFI](https://mozilla.github.io/uniffi-rs/) tool.
+
+### For contributors
+
+This section is a primer for engineers contributing code to the Search Rust component.
+
+`search.udl` describes the component's interface for foreign language consumers. UniFFI uses this file to generate the language bindings for each platform.
+
+`store.rs` contains the implementation of `SuggestStore` and most of the Suggest component's tests.
+
+## Documentation
+
+Each Rust file contains [inline documentation](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html) for types, traits, functions, and methods.
+
+Documentation for `pub` symbols is written with application developers in mind: even if you're a Desktop, Android or iOS developer, the Rust documentation is meant to give you an understanding of how the Gecko, Kotlin and Swift bindings work.
+
+You can see the documentation for all public symbols by running from the command line:
+
+```shell
+cargo doc --open
+```
+
+By convention, symbols that are exported as part of the component's foreign language interface have `pub` visibility, and symbols that are only used within the component have `pub(crate)` or private visibility. As an exception to this convention, symbols with [doctests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) are also `pub`, because doctests [can only link against public symbols](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#include-items-only-when-collecting-doctests).
+
+If you're working on the component, you can see the documentation for `pub(crate)` and private symbols using:
+
+```shell
+cargo doc --document-private-items --open
+```
+
+💡 If you're adding a new suggestion type, the documentation for the `rs` module is a great place to start.
+
+Please help us keep our documentation useful for everyone, and feel free to file bugs for anything that looks unclear or out-of-date!
+
+## Tests
+
+Tests are run with
+
+```shell
+cargo test -p search
+```
+
+## ⚠️ Breaking Changes ⚠️
+
+A "breaking change" is any code change that breaks the build, tests, or behavior of a consuming application.
+
+These are some common changes that can break consumers:
```

Comment:
I wonder if we should nix this list here (and in the Suggest readme), and expand on it in our ["breaking changes"](https://mozilla.github.io/application-services/book/howtos/breaking-changes.html) page?

I think I remember a message from @0c0w3 (please forgive me if I'm misremembering!) that this list was extremely confusing, and I can definitely see that rereading it a few months later!

Reply:
I've dropped it from here for now. I'll let you manage the other page, since I think you understand more about it.